src/components/UpdateSubscription/Unsubscribe.tsx

```diff
@@ -31,33 +29,28 @@ const Unsubscribe = ({
     scheduledSwitch,
     setCheckedReason
   } = useUnsubscribe(skipCancellationSurveyStep);
+
+  const { updateSubscription } = useAppSelector(selectPopupManager);
+  const offerId = updateSubscription?.offerData?.offerId || '';
+  const { data: allSwitchSettings } = useAppSelector(selectSwitchSettings);
+  const availableDowngrades =
```

Comment:
please extract all the downgrades logic to a custom hook. this way the component will be easier to maintain, understand and test. additionally, we'll be better prepared for the upcoming proper pause feature project

Reply:
done. Additionaly downgrads are now properly filtered